project-rome-docs/objectivec-api/remotesystems.commanding/MCDAppServiceConnection.md

```diff
@@ -0,0 +1,114 @@
+```
+This class manages a connection to a particular remote app service.
+
+## Properties
+
+### appServiceInfo
+`@property(nonatomic, retain, nonnull) MCDAppServiceInfo* appServiceInfo;`
+
+Provides details about the target app service to connect to.
+
+### requestReceived 
+`@property(nonatomic, readonly, nonnull) MCDEvent<MCDAppServiceConnection*, MCDAppServiceRequestReceivedEventArgs*>* requestReceived;`
+
+Event for when a service request is received from a remote app.
+
+### serviceClosed 
+`@property(nonatomic, readonly, nonnull) MCDEvent<MCDAppServiceConnection*, MCDAppServiceClosedEventArgs*>* serviceClosed;`
+
+Event for when connection to the app service closes.
+
+**Note:** On the client app, the "app service closed" event is not raised when the `close` method is called (by either the
+client or the host); it is only raised on the client app for unexpected connection interruptions. On the host app, the event _is_
+raised when the client calls `close` (as well as for unexpected interruptions).
+
+## Constructors
+
+### init
+`- (nullable instancetype)init;`
+
+Creates and initializes a new instance of this class.
+
+#### Returns
+The initialized [MCDAppServiceConnection](MCDAppServiceConnection.md) if successful, otherwise nil.
+
+### initWithAppServiceInfo
+- (nullable instancetype)initWithAppServiceInfo:(nonnull MCDAppServiceInfo*) appServiceInfo;
+
+#### Parameters
+* `appServiceInfo` The app service description.
+
+#### Returns
+Returns the initialized [MCDAppServiceConnection](MCDAppServiceConnection.md) if successful, otherwise nil.
+
+### appServiceConnectionWithAppServiceInfo
+`+ (nullable instancetype)appServiceConnectionWithAppServiceInfo:(nonnull MCDAppServiceInfo*) appServiceInfo;`
+
+Creates and initializes a new instance of this class with app service information.
+
+#### Parameters
+* `appServiceInfo` 
+
+The app service description.
+
+#### Returns
+Returns the initialized [MCDAppServiceConnection](MCDAppServiceConnection.md) if successful, otherwise nil.
+
+## Methods
+
+### openRemoteAsync
+`- (void)openRemoteAsync:(nonnull MCDRemoteSystemConnectionRequest*)connectionRequest completion:(nonnull void (^)(MCDAppServiceConnectionStatus, NSError* _Nullable))completion;`
+
+Opens the app service connection on the specified remote device or application. If the connection fails to open, an exception is thrown.
+
+>**Note:** This method will thrown an exception if any of the following are true:
+> * The connection is already open or is in the process of opening.
+> * The app service description has not been set or has been cleared.
+> * The platform has not been initialized.
+> * The app has subscribed a method to the connection's "request received" event but has not provided a **MCDNotificationProvider** when initializing the platform. All hosting scenarios require a **MCDNotificationProvider**.
+
+#### Parameters
+* `connectionRequest` 
+
+The connection request indicating which remote app service to target.
```

Comment:
>remote app service [](start = 40, length = 18)

It indicates which remote system or remote app to target, but not which remote app service.